mobile/onied/lib/blocs/authorization/authorization_bloc.dart

```diff
@@ -1,46 +1,133 @@
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:go_router/go_router.dart';
 import 'package:onied_mobile/app/config.dart';
-import 'package:onied_mobile/form_data/login_vk_form_data.dart';
-import 'package:onied_mobile/providers/vk_auth_service.dart';
+import 'package:onied_mobile/models/auth/credentials.dart';
+import 'package:onied_mobile/requests/forms/login_vk_form_data.dart';
+import 'package:onied_mobile/providers/vk_auth_provider.dart';
 import 'package:onied_mobile/repositories/user_repository.dart';
+import 'package:onied_mobile/models/enums/auth_status.dart';
+import 'package:onied_mobile/requests/refresh_request.dart';
 
 import 'authorization_bloc_event.dart';
 import 'authorization_bloc_state.dart';
 
 class AuthorizationBloc
     extends Bloc<AuthorizationBlocEvent, AuthorizationBlocState> {
   final UserRepository repository;
-  final VKAuthService vkAuthService;
+  final VKAuthProvider vkAuthProvider;
+  final GoRouter router;
 
-  AuthorizationBloc({required this.repository, required this.vkAuthService})
-    : super(LoadingState()) {
+  AuthorizationBloc({
+    required this.repository,
+    required this.vkAuthProvider,
+    required this.router,
+  }) : super(LoadingState()) {
     on<Login>(_onLogin);
     on<LoginWithVk>(_onLoginVk);
     on<Register>(_onRegister);
+    on<AppStarted>(_onAppStarted);
   }
 
   Future<void> _onLogin(
     Login event,
     Emitter<AuthorizationBlocState> emit,
   ) async {
-    repository.login(event.formData);
+    try {
```

Comment:
Может быть стоит начать загрузку?

```suggestion
    emit(LoadingState());
    try {
```